store/tikv/mock-tikv/mvcc_leveldb.go

```diff
@@ -0,0 +1,732 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mocktikv
+
+import (
+	"bytes"
+	"math"
+	"sync"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/goleveldb/leveldb"
+	"github.com/pingcap/goleveldb/leveldb/iterator"
+	"github.com/pingcap/goleveldb/leveldb/opt"
+	"github.com/pingcap/goleveldb/leveldb/storage"
+	"github.com/pingcap/goleveldb/leveldb/util"
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/tidb/util/codec"
+)
+
+// MVCCLevelDB implements the MVCCStore interface.
+type MVCCLevelDB struct {
+	// Key layout:
+	// ...
+	// Key_lock        -- (0)
+	// Key_verMax      -- (1)
+	// ...
+	// Key_ver+1       -- (2)
+	// Key_ver         -- (3)
+	// Key_ver-1       -- (4)
+	// ...
+	// Key_0           -- (5)
+	// NextKey_verMax  -- (6)
+	// ...
+	// NextKey_ver+1   -- (7)
+	// NextKey_ver     -- (8)
+	// NextKey_ver-1   -- (9)
+	// ...
+	// NextKey_0       -- (10)
+	// ...
+	// EOF
+	db *leveldb.DB
+	mu sync.RWMutex
+}
+
+var lockVer uint64 = math.MaxUint64
```

Comment:
because I want this key layout:

```
	// Key layout:
	// ...
	// Key_lock        -- (0)
	// Key_verMax      -- (1)
	// ...
	// Key_ver+1       -- (2)
	// Key_ver         -- (3)
	// Key_ver-1       -- (4)
	// ...
	// Key_0           -- (5)
	// NextKey_verMax  -- (6)
	// ...
	// NextKey_ver+1   -- (7)
	// NextKey_ver     -- (8)
	// NextKey_ver-1   -- (9)
	// ...
	// NextKey_0       -- (10)
	// ...
	// EOF
```